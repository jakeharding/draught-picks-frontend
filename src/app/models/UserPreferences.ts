/**
 * UserPreferences.ts
 *
 * Created by jake
 * Created on 2/27/18
 *
 * Model user preferences.
 */


export default class UserPreferences {
  uuid: string;
  abv_low: string;
  abv_hi: string;
  ibu_low: string;
  ibu_hi: string;
  like_description: string;
  user: string;

  constructor() {

  }
}
