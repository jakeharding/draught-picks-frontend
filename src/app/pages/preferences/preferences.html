<draught-header></draught-header>

<ion-content class="ion-padding">
  <ion-row>
    <ion-col class="below-header ion-text-center" size="12" sizeSm="10" offsetSm="1" sizeMd="6" offsetMd="3" sizeLg="6" offsetLg="3" sizeXl="4" offsetXl="4">

      <h1>Tell us about the beer you like!</h1>

      <p><i>The more you tell us the better we can recommend beers.</i></p>

      <ion-row>
        <h3>Select some of your favorites:</h3>
        <ion-auto-complete class="ion-text-left" tid="enterBeerInput"
                           [(ngModel)]="beerSearch"
                           (itemSelected)="favoriteSelected($event)"
                           [dataProvider]="beerProvider"></ion-auto-complete>
      </ion-row>

      <div *ngFor="let b of user?.favorite_beers">
        <ion-item>
          {{ b.name }}
          <ion-buttons slot="end">
            <ion-button>
              <b (click)="removeFavorite(b)">X</b>
            </ion-button>
          </ion-buttons>
        </ion-item>
      </div>

      <h2>Ranges of ABV and IBU</h2>

      <ion-row class="ion-text-center">
        <ion-col>
          <h3>ABV
              <ion-icon
                  slot="end" color="primary"
                  name="information-circle" class="infoButton" tid="abvInfoButton"
                  routerLink="./abv-info" routerDirection="forward"></ion-icon>
          </h3>
        </ion-col>

        <ion-col>
          <h3>IBU
              <ion-icon color="primary" name="information-circle"
                        class="infoButton" tid="ibuInfoButton"
                        routerLink="./ibu-info" routerDirection="forward"></ion-icon>
          </h3>
        </ion-col>
      </ion-row>

      <form [formGroup]="prefsForm" (ngSubmit)="savePrefs()">
        <ion-row class="ion-text-center">
          <ion-col class="ion-text-center">
            <ion-item>
              <ion-label color="primary" position="floating">High</ion-label>
              <ion-input type="number" formControlName="abv_hi" tid="enterAbvHigh" [(ngModel)]="prefs.abv_hi"></ion-input>
            </ion-item>
            <ion-item>
              <ion-label color="primary" position="floating">Low</ion-label>
              <ion-input type="number" formControlName="abv_low" tid="enterAbvLow" [(ngModel)]="prefs.abv_low"></ion-input>
            </ion-item>
          </ion-col>

          <ion-col class="ion-text-center">
            <ion-item>
              <ion-label color="primary" position="floating">High</ion-label>
              <ion-input type="number" formControlName="ibu_hi" tid="enterIbuHigh" [(ngModel)]="prefs.ibu_hi"></ion-input>
            </ion-item>
            <ion-item>
              <ion-label color="primary" position="floating">Low</ion-label>
              <ion-input type="number" formControlName="ibu_low" tid="enterIbuLow" [(ngModel)]="prefs.ibu_low"></ion-input>
            </ion-item>
          </ion-col>

        </ion-row>

        <ion-row class="ion-margin">
          <h2>Describe your favorite beer(s) below:
                <ion-icon color="primary" name="information-circle"
                          class="infoButton" tid="descInfoButton" routerLink="./beer-favorite-info"></ion-icon>
          </h2>
        </ion-row>

        <ion-row>
          <ion-textarea
              formControlName="like_description"
              tid="enterTextArea"
              rows="7"
              [(ngModel)]="prefs.like_description">
          </ion-textarea>
        </ion-row>
          <ion-button
              tid="submitBtn"
              [disabled]="prefsForm.invalid">
            Submit
          </ion-button>
      </form>
    </ion-col>
  </ion-row>
</ion-content>
