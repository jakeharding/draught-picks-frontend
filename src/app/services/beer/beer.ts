import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment as Env } from '../../../environments/environment';
import Beer from '../../models/Beer';
import PageResponse from '../../models/PageResponse';
import { LIMIT } from '../../directives/infinite-scroller/infinite-scroller';
import { EMPTY, Observable } from 'rxjs';
import { map } from 'rxjs/operators';
import { AutoCompleteService } from 'ionic4-auto-complete';
/*
  Generated class for the BeerProvider provider.

  See https://angular.io/guide/dependency-injection for more info on providers
  and Angular DI.
*/
@Injectable({
  providedIn: 'root',
})
export class BeerProvider implements AutoCompleteService {
  url: string;
  recentsUrl: string;
  recommendedUrl: string;
  labelAttribute = 'name'; // Attribute needed for the AutoCompleteService
  constructor(public http: HttpClient) {
    this.url = `${Env.REST_API_ROOT}/beers`;
    this.recentsUrl = `${Env.REST_API_ROOT}/recent-beers`;
    this.recommendedUrl = `${Env.REST_API_ROOT}/recommended-beers`;
  }

  /**
   * search function
   * Parameters: Observable of type Beer list
   * Sets the limit and offset for the infinite scrolling
   */
  search(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }
    return this.http.get(this.url, {params}).pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * Method called in auto complete search bar.
   * @param beerName - Name of beer
   * @returns - List of beers
   */
  getResults(beerName: string) {
    if (beerName) {
      return this.search({search: beerName});
    }
    return EMPTY;
  }

  /**
   * recents function
   * Parameters: Observable of type Beer list
   * Sets the limit and offset for the infinite scrolling
   */
  recents(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }

    return this.http.get(this.recentsUrl, {params}).pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * recommended function
   * Parameters: query params for pagination
   * Retrieves a page of recommended beers for an authenticated user.
   */
  recommended(params: any): Observable<Beer[]> {
    if (!params.limit) {
      params.limit = LIMIT;
    }
    if (!params.offset) {
      params.offset = 0;
    }
    return this.http.get(this.recommendedUrl, {params})
      .pipe(map(({results}: PageResponse<Beer>) => results));
  }

  /**
   * createRecent function
   * Parameters: Beer
   * returns the recent beer list
   */
  createRecent(recent: Beer) {
    return this.http.post(this.recentsUrl, {beer: recent.uuid}).toPromise();
  }

  /**
   * Retrieve a information about a beer.
   * @param uuid - ID of beer
   * @returns - Promise a beer
   */
  retrieve(uuid: string) {
    return this.http.get(`${this.url}/${uuid}`).toPromise();
  }


}
