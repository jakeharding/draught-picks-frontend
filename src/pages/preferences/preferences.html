<ion-header>
  <ion-navbar color="primary">
    <ion-title text-center>Draught Picks</ion-title>
  </ion-navbar>
</ion-header>

<ion-content padding>
  <ion-row>
    <ion-col text-center col-12 col-sm-10 offset-sm-1 col-md-6 offset-md-3 col-lg-6 offset-lg-3 col-xl-4 offset-xl-4>

      <h1>Tell us about the beer you like!</h1>

      <p><i>The more you tell us the better we can recommend beers.</i></p>

      <ion-item>

        <ion-input type="text" placeholder="Enter Beer"></ion-input>
      </ion-item>

      <button ion-button clear icon-start (click)="showErrorToastWithButton('top');">
        <ion-icon name='beer'></ion-icon>
        ADD ANOTHER
      </button>

      <button ion-button color="danger" clear icon-start>
        <ion-icon name='md-remove' is-active="false"></ion-icon>
        DELETE LAST BEER
      </button>


      <h2>Ranges of ABV and IBU
        <button ion-button clear icon-start>
          <ion-icon color="primary" name="information-circle"></ion-icon>
        </button>
      </h2>

      <ion-row text-center>
        <ion-col><h3>ABV</h3></ion-col>
        <ion-col><h3>IBU</h3></ion-col>
      </ion-row>

      <ion-row text-center>


        <ion-col text-center>
          <ion-item>
            <ion-label color="primary" floating>High</ion-label>
            <ion-input></ion-input>
          </ion-item>
          <ion-item>
            <ion-label color="primary" floating>Low</ion-label>
            <ion-input></ion-input>
          </ion-item>
        </ion-col>

        <ion-col text-center>
          <ion-item>
            <ion-label color="primary" floating>High</ion-label>
            <ion-input></ion-input>
          </ion-item>
          <ion-item>
            <ion-label color="primary" floating>Low</ion-label>
            <ion-input></ion-input>
          </ion-item>
        </ion-col>

      </ion-row>

      <h2>Describe Your Ideal Beer(s)</h2>


      <ion-label color="primary" stacked>Describe your favorite beer(s) below:</ion-label>

      <ion-row>
        <ion-textarea rows="7"> [(ngModel)]="msg"></ion-textarea>
      </ion-row>

      <button ion-button (click)="showErrorToastWithButton('top');">
        Submit
      </button>

    </ion-col>
  </ion-row>
</ion-content>
