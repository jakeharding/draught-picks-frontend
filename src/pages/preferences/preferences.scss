page-preferences {
  ion-textarea {
    border: 1px solid lightgray;
    margin-bottom: 15px;
  }
  ion-auto-complete {
    ul {
      width: 97%;
      left: 8px;
    }
  }

  .infoButton {
    padding: 0px;
    vertical-align: middle;
  }
}
