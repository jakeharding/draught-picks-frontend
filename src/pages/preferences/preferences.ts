import { Component } from '@angular/core';
import { IonicPage, NavController, NavParams } from 'ionic-angular';
import { ToastController } from 'ionic-angular';
import { LoginRequired } from '../../providers/auth/auth';
/**
 * Generated class for the PreferencesPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@LoginRequired
@IonicPage()
@Component({
  selector: 'page-preferences',
  templateUrl: 'preferences.html',
})
export class PreferencesPage {

  constructor(public navCtrl: NavController, public navParams: NavParams, public toastCtrl: ToastController) {
  }

  public showErrorToastWithButton(position: string) {
    console.log('Test : showErrorToastWithButton')
    let toast = this.toastCtrl.create({
      message: 'No network connection, try again later',
      duration: 3000,
      position: "top",
      cssClass: "error-toast"
    });
    toast.present();
  }
}
