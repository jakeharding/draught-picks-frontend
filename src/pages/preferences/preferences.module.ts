import { NgModule } from '@angular/core';
import { IonicPageModule } from 'ionic-angular';
import { PreferencesPage } from './preferences';
import { AutoCompleteModule } from "ionic2-auto-complete";
import { ComponentsModule } from "../../components/components.module";

@NgModule({
  declarations: [
    PreferencesPage,
  ],
  imports: [
    AutoCompleteModule,
    IonicPageModule.forChild(PreferencesPage),
    ComponentsModule,
  ],
})
export class PreferencesPageModule {}
