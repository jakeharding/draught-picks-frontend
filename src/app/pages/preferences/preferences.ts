import { Component } from '@angular/core';
import { NavController } from '@ionic/angular';
import { ToastController } from '@ionic/angular';
import { LoginRequired } from '../../services/auth/auth';
import { BeerProvider } from '../../services/beer/beer';
import Beer from '../../models/Beer';
import { UserProvider } from '../../services/user/user';
import User from '../../models/User';
import { PreferencesProvider } from '../../services/preferences/preferences';
import UserPreferences from '../../models/UserPreferences';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { BeerFavoriteInfoPage } from '../beer-favorite-info/beer-favorite-info';
import { AbvInfoPage } from '../abv-info/abv-info';
import { IbupagePage } from '../ibupage/ibupage';
import { BasePage } from '../BasePage';

/**
 * Generated class for the PreferencesPage page.
 *
 * See https://ionicframework.com/docs/components/#navigation for more info on
 * Ionic pages and navigation.
 */

@LoginRequired
@Component({
  selector: 'page-preferences',
  templateUrl: 'preferences.html',
})
export class PreferencesPage extends BasePage {

  beerResults: Array<Beer>;
  beerSearch: string;
  user: User;
  prefs: UserPreferences;
  prefsForm: FormGroup;
  goToBeerFavoriteInfoPage: any;
  goToABVPage: any;
  goToIBUPage: any;

  /**
   * PreferencesPage constructor
   * Required constructor for the PreferencesPage class
   * Sets the beerSearch, goToBeerFavoriteInfoPage, goToABVPage, goToIBUPage, and userProvider values accordingly
   */
  constructor(public navCtrl: NavController,
              public toastCtrl: ToastController, public beerProvider: BeerProvider,
              private userProvider: UserProvider, private preferencesProvider: PreferencesProvider,
              public formBuilder: FormBuilder) {
    super('preferences');
    this.beerSearch = '';
    this.goToBeerFavoriteInfoPage = BeerFavoriteInfoPage;
    this.goToABVPage = AbvInfoPage;
    this.goToIBUPage = IbupagePage;
    this.userProvider.retrieve().then( (user: User) => {
      this.user = user;
    });

    this.prefs = {
      uuid: '',
      abv_low: '',
      abv_hi: '',
      ibu_low: '',
      ibu_hi: '',
      like_description: '',
      user: '',
    } as UserPreferences;

    this.preferencesProvider.retrieve().then((prefs: UserPreferences) => {
      if (prefs) {
        this.prefs = prefs;
      }
    });

    this.prefsForm = this.formBuilder.group({
      abv_low: [this.prefs.abv_low, [Validators.pattern('\\d+')]],
      abv_hi: [this.prefs.abv_hi, [Validators.pattern('\\d+')]],
      ibu_low: [this.prefs.ibu_low, [Validators.pattern('\\d+')]],
      ibu_hi: [this.prefs.ibu_hi, [Validators.pattern('\\d+')]],
      like_description: [this.prefs.like_description, []]
    });
  }

  /**
   * savePrefs function
   * No Parameters
   * Gets the user preferences and saves them in the database
   */
  public async savePrefs() {
    // TODO Use ToastProvider here.
    const toast = await this.toastCtrl.create({
      message: 'Your recommendations will be here soon!',
      duration: 3000,
      position: 'top',
      cssClass: 'success-toast'
    });
    await toast.present();

    this.prefsForm.value.abv_low = this.prefsForm.value.abv_low === '' ? null : this.prefsForm.value.abv_low;
    this.prefsForm.value.abv_hi = this.prefsForm.value.abv_hi === '' ? null : this.prefsForm.value.abv_hi;
    this.prefsForm.value.ibu_hi = this.prefsForm.value.ibu_hi === '' ? null : this.prefsForm.value.ibu_hi;
    this.prefsForm.value.ibu_low = this.prefsForm.value.ibu_low === '' ? null : this.prefsForm.value.ibu_low;

    this.preferencesProvider.save(
      Object.assign({}, this.prefs, this.prefsForm.value, {user: this.user.uuid})).then(async (prefs: UserPreferences) => {
      this.prefs = prefs;
      const t = await this.toastCtrl.create({
        message: 'Your recommendations have arrived! Go to the home tab to see them!',
        duration: 3000,
        position: 'top',
        cssClass: 'success-toast'
      });
      await t.present();
    }, async () => {
      const t = await this.toastCtrl.create({
        message: 'We couldn\'t save your profile at the moment. Please try again.',
        duration: 3000,
        position: 'top',
        cssClass: 'error-toast'
      });
      await t.present();
    });

  }

  /**
   * Called when a user selects a favorite.
   * @param beer - Beer object
   */
  favoriteSelected(beer: Beer) {
    if (this.user.favorite_beers.indexOf(beer) < 0) {
      this.user.favorite_beers.push(beer);
      this.userProvider.update(this.user).then((user: User) => {
        this.user.favorite_beers = user.favorite_beers;
      });
    }
    this.beerSearch = '';
  }

  /**
   * Called when a user removes a beer from their favorites.
   * @param beer - beer object
   */
  removeFavorite(beer: Beer) {
    this.user.favorite_beers = this.user.favorite_beers.filter((b: Beer) => {
      return b.uuid !== beer.uuid;
    });
    this.userProvider.update(this.user);
  }

  /**
   * Called in the autocomplete in response to entry
   * @param event - Event object
   */
  search(event: Event) {
    if (this.beerSearch && this.beerSearch.length > 0) {
      this.beerProvider.search(this.beerSearch).toPromise().then((results: Array<Beer>) => {
        // TODO: This would be better used as an observable.
        this.beerResults = results;
      });
    }
  }
}
