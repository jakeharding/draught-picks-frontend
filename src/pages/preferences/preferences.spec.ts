/**
 * preferences.spec.ts
 *
 * Created by jake
 * Created on 11/16/18
 *
 * Test the PreferencesPage.
 */
import { PreferencesPage } from "./preferences";
import { async, ComponentFixture, TestBed } from "@angular/core/testing";
import { CUSTOM_ELEMENTS_SCHEMA } from "@angular/core";
import { NavController, NavParams, NavPush, ToastController } from "ionic-angular";
import { BeerProvider } from "../../providers/beer/beer";
import { UserProvider } from "../../providers/user/user";
import { PreferencesProvider } from "../../providers/preferences/preferences";
import { FormBuilder, FormGroupDirective } from "@angular/forms";
import { mockToast, mockToastController } from '../../jestGlobalMocks';
import User from "../../models/User";
import UserPreferences from "../../models/UserPreferences";
import Beer from "../../models/Beer";
import { Observable } from "rxjs";

describe('PreferencesPage', () => {
  let component: PreferencesPage;
  let fixture: ComponentFixture<PreferencesPage>;

  const mockBeer = {uuid: 'beerId'} as Beer;
  const mockUser = {favorite_beers: [mockBeer]} as User;
  const mockUserProvider = {
    retrieve: jest.fn(() => Promise.resolve({} as User)),
    update: jest.fn(() => Promise.resolve(mockUser))
  };
  const mockPrefsProvider = {
    retrieve: jest.fn(() => Promise.resolve({} as UserPreferences)),
    save: jest.fn(() => Promise.resolve())
  };

  const mockBeerProvider = {
    search: jest.fn()
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [CUSTOM_ELEMENTS_SCHEMA],
      declarations: [PreferencesPage, NavPush, FormGroupDirective],
      providers: [
        {provide: BeerProvider, useValue: mockBeerProvider},
        {provide: UserProvider, useValue: mockUserProvider},
        {provide: PreferencesProvider, useValue: mockPrefsProvider},
        {provide: NavController, useValue: {}},
        {provide: NavParams, useValue: {}},
        {provide: ToastController, useValue: mockToastController},
        FormBuilder
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PreferencesPage);
    component = fixture.debugElement.componentInstance;
  });

  it('should initialize the PreferencesPage', () => {
    expect(component).toBeDefined();
    expect(fixture).toMatchSnapshot();
  });

  test('properties are set during initialization of component', () => {
    expect(mockUserProvider.retrieve).toHaveBeenCalledTimes(1);
    expect(mockPrefsProvider.retrieve).toHaveBeenCalledTimes(1);
    expect(component.user).toBeDefined();
    expect(component.prefs).toBeDefined();
    expect(component.prefsForm).toBeDefined();
  });

  test('savePrefs should call the toastController.create, prefsProvider.save, and set the prefsForm values to null',
    async () => {
    await component.savePrefs();
    expect(mockToastController.create).toHaveBeenCalledTimes(2);
    expect(mockToast.present).toHaveBeenCalledTimes(2);
    expect(mockPrefsProvider.save).toHaveBeenCalledTimes(1);
    expect(component.prefsForm.value.abv_low).toBeNull();
    expect(component.prefsForm.value.abv_hi).toBeNull();
    expect(component.prefsForm.value.ibu_low).toBeNull();
    expect(component.prefsForm.value.ibu_hi).toBeNull();
  });

  test('savePrefs should call the toastController.create, prefsProvider.save, and not change the prefsForm values',
    async () => {
    component.prefsForm.value.abv_low = '1';
    component.prefsForm.value.abv_hi = '10';
    component.prefsForm.value.ibu_low = '10';
    component.prefsForm.value.ibu_hi = '10';
    await component.savePrefs();
    expect(mockToastController.create).toHaveBeenCalledTimes(2);
    expect(mockToast.present).toHaveBeenCalledTimes(2);
    expect(mockPrefsProvider.save).toHaveBeenCalledTimes(1);
    expect(component.prefsForm.value.abv_low).toBe('1');
    expect(component.prefsForm.value.abv_hi).toBe('10');
    expect(component.prefsForm.value.ibu_low).toBe('10');
    expect(component.prefsForm.value.ibu_hi).toBe('10');
  });

  test('savePrefs should display an error toast when prefsProvider.save is rejected', async () => {
    mockPrefsProvider.save.mockImplementation(() => Promise.reject('error'));
    await component.savePrefs();
    expect(mockToastController.create).toHaveBeenCalledTimes(2);
    expect(mockToast.present).toHaveBeenCalledTimes(2);
    expect(mockPrefsProvider.save).toHaveBeenCalledTimes(1);
  });

  test("favoriteSelected should call userProvider.update with the new beer in the user's favorite", async () => {
    component.user = {favorite_beers: [] as Array<Beer>} as User;
    await component.favoriteSelected(mockBeer);
    expect(mockUserProvider.update).toHaveBeenCalledTimes(1);
    expect(component.user.favorite_beers).toContain(mockBeer);
    expect(component.beerSearch).toBe('');
  });

  test('favoriteSelected should only reset the beerSearch if the beer is already a favorite', () => {
    component.user = mockUser;
    component.favoriteSelected(mockBeer);
    expect(mockUserProvider.update).not.toHaveBeenCalled();
    expect(component.beerSearch).toBe('');
  });

  test('removeFavorite should call remove the beer from favorite beers and call userProvider.update', () => {
    component.user.favorite_beers = [mockBeer];
    component.removeFavorite(mockBeer);
    expect(component.user.favorite_beers).not.toContain(mockBeer);
    expect(mockUserProvider.update).toHaveBeenCalledTimes(1);
  });

  test('search should call beerProvider.search when a value is entered', async () => {
    const mockSearchResult = Observable.of([mockBeer]);
    mockBeerProvider.search.mockReturnValue(mockSearchResult);
    component.beerSearch = 'aBeer';
    await component.search({} as Event);
    expect(mockBeerProvider.search).toHaveBeenCalledTimes(1);
    expect(mockBeerProvider.search).toHaveBeenCalledWith(component.beerSearch);

  });
});
