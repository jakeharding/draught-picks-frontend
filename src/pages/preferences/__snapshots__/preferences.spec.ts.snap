// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreferencesPage should initialize the PreferencesPage 1`] = `
<page-preferences
  beerProvider={[Function Object]}
  beerSearch=""
  formBuilder={[Function FormBuilder]}
  goToABVPage={[Function Function]}
  goToBeerFavoriteInfoPage={[Function Function]}
  goToIBUPage={[Function Function]}
  navCtrl={[Function Object]}
  navParams={[Function Object]}
  preferencesProvider={[Function Object]}
  prefs={[Function Object]}
  prefsForm={[Function FormGroup]}
  toastCtrl={[Function Object]}
  user={[Function Object]}
  userProvider={[Function Object]}
>
  <draught-header />
  
  <ion-content
    padding=""
  >
    
    
    <ion-row>
      
      
      <ion-col
        class="below-header"
        col-12=""
        col-lg-6=""
        col-md-6=""
        col-sm-10=""
        col-xl-4=""
        offset-lg-3=""
        offset-md-3=""
        offset-sm-1=""
        offset-xl-4=""
        text-center=""
      >
        
  
        
        <h1>
          Tell us about the beer you like!
        </h1>
        
  
        
        <p>
          <i>
            The more you tell us the better we can recommend beers.
          </i>
        </p>
        
  
        
        <ion-row>
          
          
          <h3>
            Select some of your favorites:
          </h3>
          
          
          <ion-auto-complete
            tid="enterBeerInput"
          />
          
        
        </ion-row>
        
  
        
        
        
  
        
        <h2>
          Ranges of ABV and IBU
        </h2>
        
  
        
        <ion-row
          text-center=""
        >
          
          
          <ion-col>
            
            
            <h3>
              ABV
              
              <button
                class="infoButton"
                clear=""
                icon-start=""
                ion-button=""
                tid="abvInfoButton"
              >
                
                
                <ion-icon
                  color="primary"
                  name="information-circle"
                />
                
              
              </button>
              
            
            </h3>
            
          
          </ion-col>
          
  
          
          <ion-col>
            
            
            <h3>
              IBU
              
              <button
                class="infoButton"
                clear=""
                icon-start=""
                ion-button=""
                tid="ibuInfoButton"
              >
                
                
                <ion-icon
                  color="primary"
                  name="information-circle"
                />
                
              
              </button>
              
            
            </h3>
            
          
          </ion-col>
          
        
        </ion-row>
        
  
        
        <form>
          
          
          <ion-row
            text-center=""
          >
            
            
            <ion-col
              text-center=""
            >
              
              
              <ion-item>
                
                
                <ion-label
                  color="primary"
                  floating=""
                >
                  High
                </ion-label>
                
                
                <ion-input
                  formcontrolname="abv_hi"
                  tid="enterAbvHigh"
                  type="number"
                />
                
              
              </ion-item>
              
              
              <ion-item>
                
                
                <ion-label
                  color="primary"
                  floating=""
                >
                  Low
                </ion-label>
                
                
                <ion-input
                  formcontrolname="abv_low"
                  tid="enterAbvLow"
                  type="number"
                />
                
              
              </ion-item>
              
            
            </ion-col>
            
  
            
            <ion-col
              text-center=""
            >
              
              
              <ion-item>
                
                
                <ion-label
                  color="primary"
                  floating=""
                >
                  High
                </ion-label>
                
                
                <ion-input
                  formcontrolname="ibu_hi"
                  tid="enterIbuHigh"
                  type="number"
                />
                
              
              </ion-item>
              
              
              <ion-item>
                
                
                <ion-label
                  color="primary"
                  floating=""
                >
                  Low
                </ion-label>
                
                
                <ion-input
                  formcontrolname="ibu_low"
                  tid="enterIbuLow"
                  type="number"
                />
                
              
              </ion-item>
              
            
            </ion-col>
            
  
          
          </ion-row>
          
  
          
          <ion-row
            margin=""
          >
            
            
            <h2>
              Describe your favorite beer(s) below:
                
              <button
                class="infoButton"
                clear=""
                icon-start=""
                ion-button=""
                tid="descInfoButton"
              >
                
                  
                <ion-icon
                  color="primary"
                  name="information-circle"
                />
                
               
              </button>
              
            
            </h2>
            
          
          </ion-row>
          
  
          
          <ion-row>
            
            
            <ion-textarea
              formcontrolname="like_description"
              rows="7"
              tid="enterTextArea"
            >
              
            
            </ion-textarea>
            
          
          </ion-row>
          
            
          <button
            ion-button=""
            tid="submitBtn"
          >
            
              Submit
            
          </button>
          
        
        </form>
        
      
      </ion-col>
      
    
    </ion-row>
    
  
  </ion-content>
  
</page-preferences>
`;
